### DISCUSSION TITLE
How to switch between native and web screens in v8? (hybrid mobile app)

### DISCUSSION THREAD

**Discussion by Krisss22** [appium/java-client#1774] [Q&A] [answered]
Hi all, 

In Appium v7 I was successfully using ContextHandles to switch between Native and Web views in a Hybrid mobile app. 
Right now, **AppiumDriver doesn't recognize context handles:** 

So these no longer accurate 
[- Set<String> contextNames = driver.getContextHandles();](https://appium.io/docs/en/commands/context/get-contexts/index.html)
[- String context = driver.getContext();](https://appium.io/docs/en/commands/context/get-contexts/)

What other solution exist for Version 8 to switch between screens?

**Comment by valfirst** [comment, marked as answer]
[SupportsContextSwitching](https://github.com/appium/java-client/blob/f1cc1030b21b7303b7ea4bc2687618e2d4fa4291/src/main/java/io/appium/java_client/remote/SupportsContextSwitching.java) describes the context management API. 2 classes implement this interface: [AndroidDriver](https://github.com/appium/java-client/blob/938a4db16b9f3f3c724b2f5429c6670465ef41c4/src/main/java/io/appium/java_client/android/AndroidDriver.java) and [IOSDriver](https://github.com/appium/java-client/blob/938a4db16b9f3f3c724b2f5429c6670465ef41c4/src/main/java/io/appium/java_client/ios/IOSDriver.java).

**Reply by valfirst** [reply]
More details can be found in the [migration guide](https://github.com/appium/java-client/blob/master/docs/v7-to-v8-migration-guide.md#appiumdriver).

**Comment by Krisss22** [comment]
> More details can be found in the [migration guide](https://github.com/appium/java-client/blob/master/docs/v7-to-v8-migration-guide.md#appiumdriver).

does this mean I have to re-write test using Android driver instead of Appium driver?

**Reply by valfirst** [reply]
The implementation of your tests is up to you, you can choose any approach

**Reply by Krisss22** [reply]
Thank you for your help!
